### PR TITLE
Remote Substreams docs

### PR DESCRIPTION
**Pull request by dimaMachina**
- [x] simplify `_app.tsx`
- [x] simplify `pages/[locale]/index.mdx` (use better approach than proposed in https://github.com/graphprotocol/docs/issues/344)
- [x] setup remote MDX for https://substreams.streamingfast.io/  https://github.com/streamingfast/substreams/tree/develop/docs 

> ⚠️ Sometimes we can get API rate limit from `https://api.github.com/repos/streamingfast/substreams/git/trees/develop?recursive=1` (URL for getting all available docs files).
>
> Let's test it for now as is and in the future we can fetch this data only once while building our app
> ```text
> error - Error: {
>  "message": "API rate limit exceeded for 37.65.12.11. (But here's the good news: Authenticated requests get a higher rate > limit. Check out the documentation for more details.)",
>  "documentation_url": "https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"
> }
> ```

### CODE REVIEW

File: website/pages/[locale]/index.mdx [hunk 1]
```diff
@@ -1,3 +1,54 @@
...
+
+## ${t('index.products.title')}
+
+<Products />
+
+## ${t('index.supportedNetworks.title')}
+
+${t('index.supportedNetworks.description')}
+
+<SupportedNetworks />`
```
Comment (B2o5T) [inline website/pages/[locale]/index.mdx:44]: this looks pretty, + we avoid ugly code to inject frontmatter and headings for TOC

File: website/components/_app.tsx [hunk 2]
```diff
@@ -62,53 +57,9 @@ const DefaultSeoWithLocale = () => {
...
-      (pageItem): pageItem is Folder => pageItem.kind === 'Folder' && pageItem.name === locale
-    )!.children
-    pageMap.find((pageItem): pageItem is MetaJsonFile => pageItem.kind === 'Meta')!.data.index = translate(
-      translations,
-      locale,
-      'index.title'
-    )
-    return pageMap
-  }, [__nextra_internal__.pageMap, hasMDXPage, locale])
+const localeSwitcher = <LocaleSwitcher key="localeSwitcher" />
```
Comment (B2o5T) [inline website/components/_app.tsx:60]: we can avoid useMemo and just render component outside function declaration

File: website/package.json [hunk 3]
```diff
@@ -20,12 +19,13 @@
...
     "mixpanel-browser": "^2.46.0",
-    "next": "^13.3.0",
+    "next": "^13.3.1",
     "next-seo": "^6.0.0",
     "next-sitemap": "^4.0.7",
-    "nextra": "2.3.0",
+    "nextra": "2.4.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "sharp": "^0.32.0",
```
Comment (B2o5T) [inline website/package.json:28]: image

File: website/package.json [hunk 4]
```diff
@@ -9,8 +9,7 @@
     "dev": "next",
     "export": "rm -rf out && next export -o ../out/docs",
     "postbuild": "next-sitemap --config next-sitemap.config.cjs && node scripts/sitemap-ci.js",
-    "prebuild": "pnpm --filter @graphprotocol/nextra-theme build",
-    "pretypecheck": "pnpm prebuild",
+    "pretypecheck": "pnpm --filter @graphprotocol/nextra-theme build",
```
Comment (B2o5T) [inline website/package.json:12]: build for `nextra-theme` was run 2 times

**Review by dotansimha — APPROVED** [review]

**Review by saihaj — APPROVED** [review]

File: website/next.config.js [hunk 5]
```diff
@@ -14,23 +14,20 @@ const withNextra = nextra({
...
+    if (route && !result.includes('getStaticProps')) {
+      const banner = `
+import { getPageMap } from '@/components/get-page-map'
+
+export const getStaticProps = async context => ({
+  props: {
+    __nextra_pageMap: await getPageMap('${route.split('/')[1]}')
+  }
+})`
+      result += banner
```
Comment (benface) [inline website/next.config.js:27]: What does `banner` mean here?

File: website/pages/[locale]/index.mdx (hunk 1 above)
Comment (benface) [reply website/pages/[locale]/index.mdx:44]: Yes, this is great!

File: website/pages/en/substreams/[[...slug]].mdx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+          const GITBOOK_CALLOUT_REGEX = /{%.*?%}/
+          visit(tree, 'paragraph', (node) => {
+            for (const child of node.children) {
+              if (child.value) {
+                child.value = child.value.replace(GITBOOK_CALLOUT_REGEX, '')
+              }
+            }
+          })
+          done()
+        },
```
Comment (benface) [inline website/pages/en/substreams/[[...slug]].mdx:40]: What are these "callouts" that you're removing?

**Review by benface — APPROVED** [review]
Great work!

Looks like the substreams docs are only available in English at the moment, and they cannot be translated through Crowdin because they are not in this repo. Do we have any plans for that?

File: website/next.config.js (hunk 5 above)
Comment (B2o5T) [reply website/next.config.js:27]: the piece of code to include `getStaticProps` for pages that don't have it

File: website/pages/en/substreams/[[...slug]].mdx (hunk 6 above)
Comment (B2o5T) [reply website/pages/en/substreams/[[...slug]].mdx:40]: something like this https://docs.gitbook.com/content-creation/blocks/hint#git-sync-representation-in-markdown

> **Note**: I remove only `callout tags` not the callout itself

Comment (benface) [reply website/pages/en/substreams/[[...slug]].mdx:40]: Got it, thank you.

**Comment by benface** [comment]
@B2o5T – Something else that I noticed: the substreams pages don't set the current page's title in the `<title>`. It's always just `The Graph Docs`, as opposed to e.g. `Running Substreams - The Graph Docs`.

**Pull request merged**